proxy/http/HttpSM.cc

```diff
@@ -751,6 +751,27 @@ HttpSM::state_read_client_request_header(int event, void *data)
     milestones[TS_MILESTONE_UA_READ_HEADER_DONE] = Thread::get_hrtime();
   }
 
+#if TS_HAS_TLS_EARLY_DATA
+  SSLNetVConnection *ssl_vc = dynamic_cast<SSLNetVConnection *>(netvc);
+  if (ssl_vc != nullptr && ssl_vc->read_from_early_data) {
+    ssl_vc->read_from_early_data = false;
```

Comment:
I may misread the code, but what if the protocol is HTTP/2 and early data contains multiple requests?

https://tools.ietf.org/html/rfc8470#section-2
> Conceptually, early data is concatenated with other application data to form a single stream. This can mean that requests are entirely contained within early data or that only part of a request is early. In a multiplexed protocol, like HTTP/2 [RFC7540] or HTTP/QUIC [HQ], multiple requests might be partially delivered in early data.

Reply:
Good point, I need to think about this a bit more.

Reply:
I think this might be a showstopper. Either needs to be resolved, or we can not allow early data when H2 (or H3) is negotiated.

Reply:
disabling for h2/3 is the quick and dirty way of getting around this for now. if we want to have this in now I can make that happen.